### ISSUE TITLE
Break components into separate modules

### ISSUE THREAD

**Issue by desmondmorris** [ferryjs/ferry#7]
I think we may want to break the individual components into separate modules.  This way when building user land extensions, we can extend the existing classes.  In order to do this now, we would need to pull down the main library, which may prove expensive as this project matures.

**Comment by jfexyz** [comment]
Are you talking about the components in the ferry package? I'm not sure they need to be broken out. We'll definitely want to extend classes in userland, but requiring the ferry package can just be the way to do that.

I was trying to do just this last night (class extension), but was having troubles. I think it was babel-related however, as that adds some complexity to our module structure I haven't quite figured out yet.

**Comment by desmondmorris** [comment]
Yeah exactly.  I guess I am just worried about requiring the full lib within the extensions.  It may not be a problem though.
